app/models/service.rb

```diff
@@ -181,29 +181,35 @@ def shutdown_guest
   end
 
   def power_states_match?(action)
-    if composite? && (power_states.uniq == map_composite_power_states(action))
-      return update_power_status(action)
-    elsif atomic? && (power_states[0] == POWER_STATE_MAP[action])
-      return update_power_status(action)
-    end
+    return update_power_status(action) if all_states_match?(action)
     false
   end
 
-  def composite?
-    service_template ? service_template.composite? : children.present?
+  def all_states_match?(action)
+    return true if child_resources? && (power_states.uniq == map_power_states(action))
+    return true if empty_child_resources? && (power_states[0] == POWER_STATE_MAP[action])
+    false
   end
 
-  def atomic?
-    service_template ? service_template.atomic? : children.empty?
+  def child_resources?
+    children.present?
+  end
+
+  def empty_child_resources?
```

Comment:
Discussed with @syncrou and was originally thinking these methods could be remove, but are leaning toward replacing them with `atomic?` and `composite?` methods to match the terminology used for service_templates.

The difference with services is they can change between atomic and composite after they are created through automate or API calls.  Therefore we need methods to determine what their current configuration is.